Add NIS One-Click-Backup

dev-nis opened the pull request:
## Description of changes
Add the new application "NIS One-Click-Backup" to nixpkgs. It is a simple program to backup folders to an external location by copying them. The source code can be found [here](https://gitlab.com/dev-nis/nis-one-click-backup-qt)
<!--
For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [x] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

dev-nis committed:
one-click-backup: init at 1.2.2.1

Co-authored-by: Cosima Neidahl <opna2608@protonmail.com>

OPNA2608 commented on pkgs/applications/backup/nis-one-click-backup/default.nix:
```diff
@@ -0,0 +1,43 @@
+, extra-cmake-modules
+}:
+
+stdenv.mkDerivation rec {
```
```suggestion
stdenv.mkDerivation (finalAttrs: {
```

OPNA2608 commented on pkgs/applications/backup/nis-one-click-backup/default.nix:
```diff
@@ -0,0 +1,43 @@
+  ];
+
+   buildInputs = [
+    qt6.full
```
Please select only the parts of Qt that your package needs, `full` will pull alot of dependencies you very likely don't need.

dev-nis replied:
Well that's a problem I was already thinking about. The problem is, I'm not sure what parts my application needs. It is a QML application, but I didn't find a 'qt6.qml' or 'qt6.quick' or something like that on search.nixos.org (only qt6.qtquick3d, but that's not right, is it?), so I'm unsure what I should write here.

dev-nis replied:
Is qt6.qtbase enough for a QML application? Or is that only for Qt Widgets?

OPNA2608 replied:
For QML, try `qtdeclarative`.

![image](https://github.com/NixOS/nixpkgs/assets/23431373/da9a8b27-45f2-46dd-ba7c-d5795855e97d)

dev-nis replied:
I will try that, thanks!

dev-nis replied:
Should now be fixed with commit [64ae0e7](https://github.com/NixOS/nixpkgs/pull/287497/commits/64ae0e7aaf0da36b871fd875485fb8177437ec20)

OPNA2608 commented on pkgs/applications/backup/nis-one-click-backup/default.nix:
```diff
@@ -0,0 +1,43 @@
+  meta = with lib; {
+    description = "Simple Program to backup folders to an external location by copying them";
+    homepage = "https://gitlab.com/dev-nis/nis-one-click-backup-qt";
+    changelog = "https://gitlab.com/dev-nis/nis-one-click-backup-qt/-/blob/${src.rev}/CHANGELOG.md";
```
```suggestion
    changelog = "https://gitlab.com/dev-nis/nis-one-click-backup-qt/-/blob/${finalAttrs.version}/CHANGELOG.md";
```

OPNA2608 commented on pkgs/applications/backup/nis-one-click-backup/default.nix:
```diff
@@ -0,0 +1,43 @@
+    homepage = "https://gitlab.com/dev-nis/nis-one-click-backup-qt";
+    changelog = "https://gitlab.com/dev-nis/nis-one-click-backup-qt/-/blob/${src.rev}/CHANGELOG.md";
+    license = licenses.gpl3Only;
+    maintainers = with maintainers; [ ];
```
[`meta.maintainers` must be set for new packages.](https://github.com/NixOS/nixpkgs/blob/2565ab7b53ae6966bd08e68203a89645961849b3/pkgs/README.md#meta-attributes)

Please read [`/maintainers/README.md`](https://github.com/NixOS/nixpkgs/blob/2565ab7b53ae6966bd08e68203a89645961849b3/maintainers/README.md), and then add your details to [`/maintainers/maintainers-list.nix`](https://github.com/NixOS/nixpkgs/blob/2565ab7b53ae6966bd08e68203a89645961849b3/maintainers/maintainer-list.nix). Commit that as a separate commit, preceding everything related to this package - look at the git history of the latter for the formatting of the commit title.

Then, adjust this line by adding yourself as the maintainer.

dev-nis replied:
This should now be implemented with commit [5f21566](https://github.com/NixOS/nixpkgs/pull/287497/commits/5f2156658dc09abc546d38c65ff73661c6c0c075) .

OPNA2608 commented on pkgs/applications/backup/nis-one-click-backup/default.nix:
```diff
@@ -0,0 +1,43 @@
+    mainProgram = "one-click-backup";
+    platforms = platforms.all;
+  };
+}
```
(to match `finalAttrs` suggestion above)

```suggestion
})
```

OPNA2608 commented on pkgs/top-level/all-packages.nix:
```diff
@@ -41948,4 +41948,6 @@ with pkgs;
 
   animdl = python3Packages.callPackage ../applications/video/animdl { };
+
+  one-click-backup = qt6Packages.callPackage ../applications/backup/nis-one-click-backup { };
```
New packages should use the [`/pkgs/by-name`](https://github.com/NixOS/nixpkgs/blob/2565ab7b53ae6966bd08e68203a89645961849b3/pkgs/by-name/README.md) system instead of the old `all-packages.nix` one, unless they meet specific criteria that make them unsuitable for the former. Please move the package, and drop the `all-packages.nix` entry.

dev-nis replied:
Am I understanding this right? I move the package definition into another folder and delete everything I added from `all-packages.nix`? But where do I tell nix to use qt6Packages.callPackage then? This doesn't seem clear to me when reading the [documentation](https://github.com/NixOS/nixpkgs/blob/2565ab7b53ae6966bd08e68203a89645961849b3/pkgs/by-name/README.md) you gave me.

OPNA2608 replied:
> Am I understanding this right?

Yep.

> But where do I tell nix to use qt6Packages.callPackage then?

`libsForQt5.callPackage` / `qt6Packages.callPackage` let you import `qtbase`, `wrapQtAppsHook` etc at the top of your derivation. Whether those are from Qt5 or Qt6 will then depend on which `callPackage` you used, letting you write your derivation somewhat Qt version-agnostic.

Since you're directly using `qt6` already, `qt6Packages.callPackage` shouldn't matter.

dev-nis replied:
I'm not really understanding what you mean here. So I need to just move my derivation to another directory and then it "just works" (no need to write the `qt6packages.callPackage `anywhere)? Because every time I've used `callPackage` without the `qt6packages`, the derivation didn't build (got an error with `wrapQtAppsHook`)

OPNA2608 replied:
Use `qt6.wrapQtAppsHook`.

dev-nis replied:
Okay, but where do I write this line? I thought with the by-name structure you don't define such things anywhere?

dev-nis replied:
Okay, I think I got it now (finally). Commit [924be71](https://github.com/NixOS/nixpkgs/pull/287497/commits/924be717ca4ebc954288443bb4493e601c41bec3) should now use the `by-name` structure. An my test machine, it builds fine.

OPNA2608 commented on pkgs/applications/backup/nis-one-click-backup/default.nix:
```diff
@@ -0,0 +1,43 @@
+    cmake
+    ninja
+    wrapQtAppsHook
+    extra-cmake-modules
```
Just as a note, not a requirement for me: Should perhaps be moved to `buildInputs` (`strictDeps = true` will break this here), but most of Nixpkgs puts it in `nativeBuildInputs` and doesn't enforce `strictDeps` yet… So I guess it's okay to leave it here.

dev-nis replied:
Even though you say it's not an requirement, I still implemented your suggestion with commit [6098f80](https://github.com/NixOS/nixpkgs/pull/287497/commits/6098f80dab4a5e5bddae62a9bd10e6ef6b8e2831)

OPNA2608 reviewed (changes requested):
Welcome to Nixpkgs. :wave: 

Please adjust your commits to the commit conventions ([common](https://github.com/NixOS/nixpkgs/blob/2565ab7b53ae6966bd08e68203a89645961849b3/CONTRIBUTING.md#commit-conventions) and [`/pkgs`-specific](https://github.com/NixOS/nixpkgs/blob/2565ab7b53ae6966bd08e68203a89645961849b3/pkgs/README.md#commit-conventions), employ squashing & rebasing), and check the suggestions.

dev-nis committed:
maintainers: add NIS

Co-authored-by: Cosima Neidahl <opna2608@protonmail.com>

OPNA2608 commented on maintainers/maintainer-list.nix:
```diff
@@ -13717,6 +13717,11 @@
+  NIS = {
+    name = "NSC IT Solutions";
+    github = "dev-nis";
+    githubId = "132921300";
```
```
error: A definition for option `lib.maintainers.NIS.githubId' is not of type `null or (unsigned integer, meaning >=0)'.
```

```suggestion
    githubId = 132921300;
```

OPNA2608 commented on pkgs/by-name/on/one-click-backup/package.nix:
```diff
@@ -0,0 +1,47 @@
+    hash = "sha256-JbgmfKEpKRj/ph8dkW0IRulmAhk6Dwv/1Z4j3hUgPXk=";
+  };
+
+ # nativeBuildInputs = [
+ #   cmake
+ #   ninja
+ #   wrapQtAppsHook
+ #   extra-cmake-modules
+ # ];
```
I meant just `extra-cmake-modules`, not everything in `nativeBuildInputs`.

```suggestion
 nativeBuildInputs = [
   cmake
   ninja
   qt6.wrapQtAppsHook
 ];
```

dev-nis replied:
Oh, I misunderstood that, sorry

OPNA2608 commented on pkgs/by-name/on/one-click-backup/package.nix:
```diff
@@ -0,0 +1,47 @@
+   buildInputs = [
+    qt6.qtdeclarative
+    qt6.qtbase
+    cmake
+    ninja
+    qt6.wrapQtAppsHook
```
```suggestion
```

OPNA2608 commented on pkgs/by-name/on/one-click-backup/package.nix:
```diff
@@ -0,0 +1,47 @@
+, cmake
+, ninja
+, qt6
+#, wrapQtAppsHook
```
```suggestion
```

OPNA2608 commented on pkgs/by-name/on/one-click-backup/package.nix:
```diff
@@ -0,0 +1,47 @@
+
+stdenv.mkDerivation (finalAttrs: {
+  pname = "one-click-backup";
+  version = "1.2.2.1";
```
This version doesn't exist on the remote. The latest pushed tag is [1.2.2](https://gitlab.com/dev-nis/nis-one-click-backup-qt/-/tree/1.2.2?ref_type=tags).

```
error: builder for '/nix/store/c4v512z51l29bh34avhq53hbbjgh4ak3-source.drv' failed with exit code 1;
       last 7 log lines:
       >
       > trying https://gitlab.com/api/v4/projects/dev-nis%2Fnis-one-click-backup-qt/repository/archive.tar.gz?sha=1.2.2.1
       >   % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
       >                                  Dload  Upload   Total   Spent    Left  Speed
       >   0    32    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
       > curl: (22) The requested URL returned error: 404
       > error: cannot download source from any mirror
       For full logs, run 'nix-store -l /nix/store/c4v512z51l29bh34avhq53hbbjgh4ak3-source.drv'.
error: 1 dependencies of derivation '/nix/store/s3glb7bvls4n7snil7mjayl5wyax75gk-one-click-backup-1.2.2.1.drv' failed to build
```

dev-nis replied:
Oh, I somehow forgot to publish a new release (I only added a new tag to git). Should be fixed now: [Version 1.2.2.1](https://gitlab.com/dev-nis/nis-one-click-backup-qt/-/releases/1.2.2.1)

OPNA2608 commented on pkgs/by-name/on/one-click-backup/package.nix:
```diff
@@ -0,0 +1,42 @@
+    owner = "dev-nis";
+    repo = "nis-one-click-backup-qt";
+    rev = finalAttrs.version;
+    hash = "sha256-JbgmfKEpKRj/ph8dkW0IRulmAhk6Dwv/1Z4j3hUgPXk=";
```
```
error: hash mismatch in fixed-output derivation '/nix/store/c4v512z51l29bh34avhq53hbbjgh4ak3-source.drv':
         specified: sha256-JbgmfKEpKRj/ph8dkW0IRulmAhk6Dwv/1Z4j3hUgPXk=
            got:    sha256-F+gA+Z4gZoNJYdy28uIjqiJcwcNsyUzl6BXsiIZO0gE=
```

```suggestion
    hash = "sha256-F+gA+Z4gZoNJYdy28uIjqiJcwcNsyUzl6BXsiIZO0gE=";
```

dev-nis replied:
I knew I had to change that, but for some reason I don't understand, nix doesn't complain about this on my build machine. That's really strange.

dev-nis replied:
Does nix somehow cache the source for builds?

OPNA2608 replied:
It doesn't complain because that hash is still present on your system, from a previous/different build. Hash exists -> no need to build, just use what it points at.

Try building `one-click-backup.src` with `--check`, that should fail.

dev-nis replied:
Ah, that makes sense. Thank you for the explanation!

OPNA2608 commented on pkgs/by-name/on/one-click-backup/package.nix:
```diff
@@ -0,0 +1,42 @@
+    changelog = "https://gitlab.com/dev-nis/nis-one-click-backup-qt/-/blob/${finalAttrs.version}/CHANGELOG.md";
+    license = licenses.gpl3Only;
+    maintainers = with maintainers; [ NIS ];
+    mainProgram = "one-click-backup";
```
The installed executable is called `NIS_One-Click-Backup_Qt`.

```suggestion
    mainProgram = "NIS_One-Click-Backup_Qt";
```

OPNA2608 reviewed (changes requested):
Result of `nixpkgs-review pr 287497` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>one-click-backup</li>
  </ul>
</details>

---

![image](https://github.com/NixOS/nixpkgs/assets/23431373/1cfca6f5-058c-4735-9b56-d507dd74268c)

Launches. Please see the suggestion I added, then squash as explained earlier. Expected order & format of commits is:

- `maintainers: add NIS`
- `one-click-backup: init at 1.2.2.1`

dev-nis commented:
Squashing and rebasing is now done! I hope I have done everything correctly, this is my first time messing around with `git rebase`

dev-nis commented:
> If you can get that fixed that'd be great
If I change the code, is there some way for me to test if it builds on Darwin? If I can get it fixed, I will upload a new release, if not I will mark it as broken.

> Doesn't look squashed to me, should show 2 commits instead of 22.

That I don't understand, I explicitely used "squash" everywhere when running `git rebase -i`. Maybe I missed something, I will look into that.

OPNA2608 commented:
> If I change the code, is there some way for me to test if it builds on Darwin?

Push a commit and fetch it as a patch.

```nix
patches = [
  # Maybe fix build on Darwin
  # Remove when version > 1.2.2.1
  (fetchpatch {
    url = "https://gitlab.com/dev-nis/nis-one-click-backup-qt/-/commit/fbecc454871c79add00ff35d363b262d543cd2ed.patch";
    hash = "";
  })
];
```

(replace with the commit's rev, fill on hash based on the hash mismatch error)

dev-nis commented:
> > If I change the code, is there some way for me to test if it builds on Darwin?
> 
> Push a commit and fetch it as a patch.
> 
> ```nix
> patches = [
>   # Maybe fix build on Darwin
>   # Remove when version > 1.2.2.1
>   (fetchpatch {
>     url = "https://gitlab.com/dev-nis/nis-one-click-backup-qt/-/commit/fbecc454871c79add00ff35d363b262d543cd2ed.patch";
>     hash = "";
>   })
> ];
> ```
> 
> (replace with the commit's rev, fill in hash based on the hash mismatch error)

Hi, I can't really figure out how this works. I added your code to my `.nix`-file and changed it accordingly, but it complains `error: undefined variable 'fetchpatch'`. What do I need to change for this to work? I can't seem to find any documentation about this
![grafik](https://github.com/NixOS/nixpkgs/assets/132921300/71d531ff-a265-4c64-9b36-482d6a39e412)

OPNA2608 commented on pkgs/by-name/on/one-click-backup/package.nix:
```diff
@@ -0,0 +1,51 @@
+{ lib
+, stdenv
+, fetchFromGitLab
```
```suggestion
, fetchFromGitLab
, fetchpatch
```

OPNA2608 commented on pkgs/by-name/on/one-click-backup/package.nix:
```diff
@@ -0,0 +1,51 @@
+    platforms = platforms.all;
+  };
+
+  patches = [
+    # Maybe fix build on Darwin
+    # Remove when version > 1.2.2.1
+    (fetchpatch {
+      url = "https://gitlab.com/dev-nis/nis-one-click-backup-qt/-/commit/314607eadfd3588d0b7082e9692f03fd8262f907.patch";
+      hash = "";
+    })
+  ];
```
Move this between `src` and `nativeBuildInputs`, there's a certain order that's common for these.

```suggestion
```

OPNA2608 commented on pkgs/by-name/on/one-click-backup/package.nix:
```diff
@@ -0,0 +1,51 @@
+    hash = "sha256-F+gA+Z4gZoNJYdy28uIjqiJcwcNsyUzl6BXsiIZO0gE=";
+  };
+
+ nativeBuildInputs = [
+   cmake
+   ninja
+   qt6.wrapQtAppsHook
+ ];
```
```suggestion
  patches = [
    # Maybe fix build on Darwin
    # Remove when version > 1.2.2.1
    (fetchpatch {
      url = "https://gitlab.com/dev-nis/nis-one-click-backup-qt/-/commit/314607eadfd3588d0b7082e9692f03fd8262f907.patch";
      hash = "";
    })
  ];
  
  nativeBuildInputs = [
    cmake
    ninja
    qt6.wrapQtAppsHook
  ];
```

dev-nis commented:
If you say everything is fine, I will squash the remaining commits

OPNA2608 commented:
qt6.qtbase is currently broken on Darwin, so can't test this: https://github.com/NixOS/nixpkgs/pull/288842#issuecomment-1946423014

OPNA2608 commented:
I would expect `qtdeclarative` to depend on `qtbase`, so I'm not sure if this really fixes the situation, but I guess there's no harm in removing it.

dev-nis commented:
> I would expect `qtdeclarative` to depend on `qtbase`, so I'm not sure if this really fixes the situation, but I guess there's no harm in removing it.

Well, seems like you were right:
`       > 
       > -- Configuring incomplete, errors occurred!
       For full logs, run 'nix log /nix/store/p04f3nsmny20v3pbxxklaixjzdglqkk7-qtbase-6.6.2.drv'.
error: 1 dependencies of derivation '/nix/store/y5lfmi3v3q0kp745d4bqmfq74qjfnksj-qttools-6.6.2.drv' failed to build
error: 1 dependencies of derivation '/nix/store/8ay1bs25mlvwpsli5yrz3zadjhwzcq9k-qttranslations-6.6.2.drv' failed to build
error: 1 dependencies of derivation '/nix/store/vlg6q8z96frv0kgk4538icqrhp3qn9kl-qtbase-6.6.2.drv' failed to build
error: 1 dependencies of derivation '/nix/store/ajw3rn9jkd4irs1j4f31ihn2ny09zs6i-qtlanguageserver-6.6.2.drv' failed to build
error: 1 dependencies of derivation '/nix/store/bbfraxbls9afv2k8nlsasxvxp4g7xvg7-qtshadertools-6.6.2.drv' failed to build
error: 3 dependencies of derivation '/nix/store/cv8fy0ivd71wna8fjzp07insgzcw8sl2-qtdeclarative-6.6.2.drv' failed to build
error: 1 dependencies of derivation '/nix/store/5npx9fymd4abddwq12wm64hsb5y53w0x-one-click-backup-1.2.2.1.drv' failed to build`

Should I then just wait for this issue to get fixed? Or should I mark Darwin as broken for now?

OPNA2608 commented:
Give it a day or so and it should (hopefully) work again, fix has been merged.

OPNA2608 commented on pkgs/by-name/on/one-click-backup/package.nix:
```diff
@@ -0,0 +1,51 @@
+
+   buildInputs = [
+    qt6.qtdeclarative
+    extra-cmake-modules
```
https://github.com/NixOS/nixpkgs/pull/245900#discussion_r1430618993

> KDE's `extra-cmake-modules` is usually placed into `nativeBuildInputs`. `strictDeps` says that's wrong

dev-nis commented:
> I would expect `qtdeclarative` to depend on `qtbase`, so I'm not sure if this really fixes the situation, but I guess there's no harm in removing it.

Well, seems like you were right:
`       > 
       > -- Configuring incomplete, errors occurred!
       For full logs, run 'nix log /nix/store/p04f3nsmny20v3pbxxklaixjzdglqkk7-qtbase-6.6.2.drv'.
error: 1 dependencies of derivation '/nix/store/y5lfmi3v3q0kp745d4bqmfq74qjfnksj-qttools-6.6.2.drv' failed to build
error: 1 dependencies of derivation '/nix/store/8ay1bs25mlvwpsli5yrz3zadjhwzcq9k-qttranslations-6.6.2.drv' failed to build
error: 1 dependencies of derivation '/nix/store/vlg6q8z96frv0kgk4538icqrhp3qn9kl-qtbase-6.6.2.drv' failed to build
error: 1 dependencies of derivation '/nix/store/ajw3rn9jkd4irs1j4f31ihn2ny09zs6i-qtlanguageserver-6.6.2.drv' failed to build
error: 1 dependencies of derivation '/nix/store/bbfraxbls9afv2k8nlsasxvxp4g7xvg7-qtshadertools-6.6.2.drv' failed to build
error: 3 dependencies of derivation '/nix/store/cv8fy0ivd71wna8fjzp07insgzcw8sl2-qtdeclarative-6.6.2.drv' failed to build
error: 1 dependencies of derivation '/nix/store/5npx9fymd4abddwq12wm64hsb5y53w0x-one-click-backup-1.2.2.1.drv' failed to build`

Should I then just wait for this issue to get fixed? Or should I mark Darwin as broken for now?

> Give it a day or so and it should (hopefully) work again, fix has been merged.

Okay! How can I trigger a rebuild after a few days?

OPNA2608 commented:
@ofborg build one-click-backup one-click-backup.passthru.tests

dev-nis commented:
Seems like it still doesn't build on Darwin I will mark "Darwin" as broken as soon as I can.

dev-nis commented:
All checks pass now, if you say everything is fine, I will squash the commits again.

OPNA2608 commented on pkgs/by-name/on/one-click-backup/package.nix:
```diff
@@ -0,0 +1,43 @@
+    qt6.wrapQtAppsHook
+  ];
+
+   buildInputs = [
```
```suggestion
  buildInputs = [
```

OPNA2608 commented on pkgs/by-name/on/one-click-backup/package.nix:
```diff
@@ -0,0 +1,43 @@
+{ lib
+, stdenv
+, fetchFromGitLab
+, fetchpatch
```
Not used anymore, can go.

```suggestion
```

OPNA2608 reviewed (changes requested):
Otherwise looks fine for squashing

dev-nis commented:
Squashed now!

OPNA2608 reviewed (approved):
Result of `nixpkgs-review pr 287497` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>one-click-backup</li>
  </ul>
</details>

---

GUI lauches, LGTM.

dev-nis commented:
Great, thank's for all your comments and help!
